extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/ResourceRegistrationUtils.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.grpc.deployment;
+
+import java.util.regex.Pattern;
+
+import org.eclipse.microprofile.config.Config;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+
+class ResourceRegistrationUtils {
```

Comment:
Wondering if this should be part of quarkus "core".

@stuartwdouglas @gsmet WDYT? This is not gRPC specific.

Reply:
> It looks good.
> 
> A few missing things:
> 
> * Documentation must be updated
I see I missed the config property javadoc :( 
Do you think the guide needs to be further updated or do you mean ^
> * Quickstart also
:+1: Quickstart is the main reason for the change :)
> 
> Should we consider it as a breaking change as the location of the certs changed.
The location of certs doesn't really change. What worked before should work now **but** for the situation when someone had a different key, cert, etc on classpath on the same "classpath path" as a file for real path.

Reply:
> It looks good.
> 
> A few missing things:
> 
> * Documentation must be updated
I see I missed the config property javadoc :( 
Do you think the guide needs to be further updated or do you mean ^
> * Quickstart also
:+1: Quickstart was the trigger for the change :)
> 
> Should we consider it as a breaking change as the location of the certs changed.
The location of certs doesn't really change. What worked before should work now **but** for the situation when someone had a different key, cert, etc on classpath on the same "classpath path" as a file for real path.

Reply:
I'm not sure tbh.

`registerResourcesForProperties` looks quite weird and I wouldn't want to make it a common pattern.

`registerResourceForProperty` I'm a bit unsure of what it does exactly. I fail to see why we would do this check here and not at an higher level (and probably throw an error because the file is not available?).